### PR TITLE
cpu/cc26xx_cc13xx: use _rom_length symbol for CCFG origin

### PR DESCRIPTION
**Pull request by jeandudey**
### Contribution description

Fixes the CCFG origin on cc26x0 CPUs, it should be at end of the ROM.

### Testing procedure

- `make -C examples/hello-world menuconfig BOARD=cc2650-launchpad && make && arm-none-eabi-readelf --hex-dump=.ccfg examples/hello-world/bin/cc2650-launchpad/hello-world.elf`
- Output should be:
```
jeandudey@jean ~/D/R/e/hello-world> arm-none-eabi-readelf --hex-dump=.ccfg /home/jeandudey/Dev/RIOT/examples/hello-world/bin/cc2650-launchpad/hello-world.elf

Hex dump of section '.ccfg':
  0x0001ffa8 00008001 100082ff fdff5800 3affaff1 ..........X.:...
  0x0001ffb8 ffffffff ffffffff ffffffff ffffffff ................
  0x0001ffc8 ffffffff ffffffff ffffffff ffffffff ................
  0x0001ffd8 ffffff00 00000000 00000000 00c5c5ff ................
  0x0001ffe8 00000000 00000000 ffffffff ffffffff ................
  0x0001fff8 ffffffff ffffffff                   ........
```

### Issues/PRs references

- #14219

### CODE REVIEW

File: cpu/cc26xx_cc13xx/vectors.c [hunk 1]
```diff
@@ -158,12 +158,14 @@ ccfg_regs_t cc26xx_cc13xx_ccfg = {
...
                                         CCFG_MODE_CONF_1_TCXO_MAX_START_s,
                                         CCFG_MODE_CONF_1_TCXO_MAX_START_m)
-#else
-                        0
-#endif /* CPU_VARIANT_X2 */
                         & CCFG_SET_BITS(SET_MODE_CONF_1_ALT_DCDC_VMIN,
                                         CCFG_MODE_CONF_1_ALT_DCDC_VMIN_s,
                                         CCFG_MODE_CONF_1_ALT_DCDC_VMIN_m)
+#else
+                          CCFG_SET_BITS(SET_MODE_CONF_1_ALT_DCDC_VMIN,
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/vectors.c:165]: No & here?

File: cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld [hunk 2]
```diff
@@ -22,7 +22,8 @@ INCLUDE cortexm_rom_offset.ld
 MEMORY
 {
     rom      (rx)   : ORIGIN = _rom_start_addr + _rom_offset, LENGTH = _fw_rom_length
-    ccfg     (rx)   : ORIGIN = 0x00057FA8,                    LENGTH = 88
+    /* CCFG starts at the end of ROM */
+    ccfg     (rx)   : ORIGIN = _rom_length,                   LENGTH = 88
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld:26]: How about keeping `ROM_LEN` at 128k and then use

```suggestion
    ccfg     (rx)   : ORIGIN = _rom_length - 88,              LENGTH = 88
```

here?

This keeps the config offset handling better contained.

File: cpu/cc26xx_cc13xx/vectors.c (hunk 1 above)
Comment (jeandudey) [reply cpu/cc26xx_cc13xx/vectors.c:165]: The asigment is direct this way,

```c
    .MODE_CONF_1 =  CCFG_SET_BITS(SET_MODE_CONF_1_ALT_DCDC_VMIN,
                   ....
```

I could add the `#ifdef` block only for the `&` but seemed very weird:

```c
#ifdef CPU_VARIANT_X2
                        /* only x2 CPUs allow TCXO settings */
                          CCFG_SET_BITS(SET_MODE_CONF_1_TCXO_TYPE,
                                        CCFG_MODE_CONF_1_TCXO_TYPE_s,
                                        CCFG_MODE_CONF_1_TCXO_TYPE_m)
                        & CCFG_SET_BITS(SET_MODE_CONF_1_TCXO_MAX_START,
                                        CCFG_MODE_CONF_1_TCXO_MAX_START_s,
                                        CCFG_MODE_CONF_1_TCXO_MAX_START_m)
                        &
#else
                          CCFG_SET_BITS(SET_MODE_CONF_1_ALT_DCDC_VMIN,
                                        CCFG_MODE_CONF_1_ALT_DCDC_VMIN_s,
                                        CCFG_MODE_CONF_1_ALT_DCDC_VMIN_m)
#endif /* CPU_VARIANT_X2 */
```

Comment (benpicco) [reply cpu/cc26xx_cc13xx/vectors.c:165]: Ok

File: cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld (hunk 2 above)
Comment (jeandudey) [reply cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld:26]: Won't it cause `rom` to overlap with `ccfg` :eyes:?

Comment (benpicco) [reply cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld:26]: Shouldn't the linker warn about overlapping sections? That would then be a build error.

Comment (jeandudey) [reply cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld:26]: >overlaps section seems to be a valid linker error string

TIL :smile: , the test blob on cc26x0 failed succesfully with a 128K blob and the .ccfg section :relieved:

Comment (jeandudey) [reply cpu/cc26xx_cc13xx/ldscripts/cc26xx_cc13xx.ld]: Fixed at https://github.com/RIOT-OS/RIOT/pull/15126/commits/ed2e04904b35ba5ca0f72fdbfd5ffc25851c55cc . Thanks for pointing it! Much cleaner

**Review by benpicco — APPROVED** [review]

**Comment by jeandudey** [comment]
Thanks for the review! Sorry for the mistake :)

**Comment by benpicco** [comment]
No worries,  thank you for the quick fix!

**Pull request merged**